packages/components/Button/src/useButton.ts

```diff
@@ -11,7 +11,7 @@ export const useButton = (props: ButtonProps): ButtonState => {
   // GH #1336: Set focusRef to null if button is disabled to prevent getting keyboard focus.
   const focusRef = disabled ? null : componentRef;
   const onClickWithFocus = useOnPressWithFocus(focusRef, onClick);
-  const pressable = useAsPressable({ ...rest, onPress: onClickWithFocus });
+  const pressable = useAsPressable({ ...rest, disabled, onPress: onClickWithFocus });
   const onKeyUpProps = useKeyProps(onClick, ' ', 'Enter');
   const isDisabled = !!disabled || !!loading;
```

Comment:
we can end up with disabled != isDisabled; should we evaluate isDisabled sooner and use it everywhere, or is there a nuanced need for the conflict to be possible?

Reply:
Yeah, we should move isDisabled up and pass it in instead of just disabled.

Reply:
I updated this in the newest iteration.